src/resdk/resources/predictions.py

```diff
@@ -0,0 +1,238 @@
+
+        :param resolwe: Resolwe instance
+        :param model_data: Resource model data
+        """
+        self.logger = logging.getLogger(__name__)
+        super().__init__(resolwe, **model_data)
+
+    def __repr__(self):
+        """Return user friendly string representation."""
+        return f"PredictionGroup <name: {self.name}>"
+
+
+class PredictionField(BaseResource):
+    """Resolwe PredictionField resource."""
+
+    endpoint = "prediction_field"
+
+    READ_ONLY_FIELDS = BaseResource.READ_ONLY_FIELDS + (
+        "description",
+        "group",
+        "label",
+        "name",
+        "sort_order",
+        "type",
+        "validator_regex",
+        "vocabulary",
+        "required",
+        "version",
+    )
+
+    def __init__(self, resolwe: "Resolwe", **model_data):
+        """Initialize the instance.
+
+        :param resolwe: Resolwe instance
+        :param model_data: Resource model data
+        """
+        self.logger = logging.getLogger(__name__)
+        #: prediction group
+        self._group = None
+        super().__init__(resolwe, **model_data)
+
+    @property
+    def group(self) -> PredictionGroup:
+        """Get prediction group."""
+        assert (
+            self._group is not None
+        ), "PredictionGroup must be set before it can be used."
+        return self._group
+
+    @group.setter
+    def group(self, payload: dict):
+        """Set prediction group."""
+        if self._group is None:
+            self._resource_setter(payload, PredictionGroup, "_group")
+        else:
+            raise AttributeError("PredictionGroup is read-only.")
+
+    def __repr__(self):
+        """Return user friendly string representation."""
+        return f"PredictionField <path: {self.group.name}.{self.name}>"
+
+    def __str__(self):
+        """Return full path of the prediction field."""
+        return f"{self.group.name}.{self.name}"
+
+
+class PredictionValue(BaseResource):
+    """Resolwe PredictionValue resource."""
+
+    endpoint = "prediction_value"
+
+    READ_ONLY_FIELDS = BaseResource.READ_ONLY_FIELDS + ("label",)
+
+    UPDATE_PROTECTED_FIELDS = BaseResource.UPDATE_PROTECTED_FIELDS + ("field", "sample")
+
+    WRITABLE_FIELDS = BaseResource.WRITABLE_FIELDS + ("value",)
+
+    def __init__(self, resolwe: "Resolwe", **model_data):
+        """Initialize the instance.
+
+        :param resolwe: Resolwe instance
+        :param model_data: Resource model data
+        """
+        self.logger = logging.getLogger(__name__)
+
+        #: prediction field
+        self._field: Optional[PredictionField] = None
+        self._value: Optional[Union[ScorePredictionType, ClassPredictionType]] = None
+        self.field_id: Optional[int] = None
+
+        #: sample
+        self.sample_id: Optional[int] = None
+        self._sample: Optional[Sample] = None
+        super().__init__(resolwe, **model_data)
+
+    @property
+    @assert_object_exists
+    def modified(self):
```

Comment:
Since it is actually the modification date from the user perspective.

When new prediction / annotation is created (that overrides the old one) it can be seen as a modification of an old value. Not sure what is the `correct`  way of handling this, I can also just leave `created` field and remove `modified`.